File: sevenkplus/normal/NA/102.rb
#include <bits/stdc++.h>

using namespace std;

typedef long long ll;
typedef pair<int, int> PII;
typedef double db;

#define fi first
#define se second
#define mp make_pair
#define pb push_back
#define pct __builtin_popcount

#define N 1010
int n, a[N], b[N];
int A, B;

int main() {
	cin >> n >> A >> B;
	set<PII> AA, BB;
	for (int i = 0; i < n; i ++) {
		cin >> a[i] >> b[i];
		a[i] = a[i]*A/100;
		AA.insert(mp(-a[i], i));
	}
	int c = A, t = 0, w = 0;
	const int ma = A+n;
	vector<PII> S;
	while (t < ma) {
		c -= w;
		c += B;
		c = min(c, A);
		if (c <= 0) {
			break;
		}
		
		while (!AA.empty() && -AA.begin()->fi >= c) {
			int x = AA.begin()->se;
			AA.erase(AA.begin());
			BB.insert(mp(-b[x], x));
		}

		if (!BB.empty()) {
			int x = BB.begin()->se;
			BB.erase(BB.begin());
			S.pb(mp(t, x));
			w += b[x];
		}

		t++;
	}
	if (t >= ma) {
		puts ("NO");
	} else {
		puts ("YES");
		printf ("%d %d\n", t, (int) S.size());
		for (auto p : S) {
			printf("%d %d\n", p.fi, p.se + 1);
		}
	}
	return 0;
}

File: origenes/normal/NA/102.cpp
#include <bits/stdc++.h>

using namespace std;
typedef long long ll;
typedef pair<int, int> pii;
#define FOR(i, a, b) for (int (i) = (a); (i) <= (b); (i)++)
#define ROF(i, a, b) for (int (i) = (a); (i) >= (b); (i)--)
#define REP(i, n) FOR(i, 0, (n)-1)
#define sqr(x) ((x) * (x))
#define all(x) (x).begin(), (x).end()
#define reset(x, y) memset(x, y, sizeof(x))
#define uni(x) (x).erase(unique(all(x)), (x).end());
#define BUG(x) cerr << #x << " = " << (x) << endl
#define pb push_back
#define eb emplace_back
#define mp make_pair
#define _1 first
#define _2 second
#define chkmin(a, b) a = min(a, b)
#define chkmax(a, b) a = max(a, b)

int n, k, a, b;
ll mmax, mmin;

void solve (int r) {
  REP(i, n) {
    if (!r && !i) continue;
    ll now = ll(n) * k / __gcd(ll(n) * k, ll(i) * k + r);
    chkmin(mmin, now);
    chkmax(mmax, now);
  }
  if (!r) mmin = 1;
}

int main() {
  scanf("%d%d%d%d", &n, &k, &a, &b);
  mmax = 1;
  mmin = 1e15;
  solve(a + b);
  solve((k - b + a) % k);
  solve((k - a + b) % k);
  solve((2 * k - a - b) % k);
  printf("%lld %lld", mmin, mmax);
}